bin/build-caverunner-pipeline.groovy

```diff
@@ -0,0 +1,71 @@
+pipeline {
+  agent { label 'sidious' }
+  options {
+    timestamps()
+    buildDiscarder(logRotator(artifactDaysToKeepStr: '10', artifactNumToKeepStr: '20', daysToKeepStr: '20', numToKeepStr: '20'))
+    disableConcurrentBuilds()
+  }
+  environment {
+    JENKINS_BUILD = 'true'
+  }
+  parameters {
+    string(name: 'CR_BRANCH', defaultValue: "*/master-v21.0", description: 'Cave Runner Branch and/or tag to use to compile. (Git tag but must be formatted as "refs/tags/{tagName}")')
+    string(name: 'sdk_build_number', defaultValue: '', description: 'Build number for SDK-Framework to use for artifacts.')
+  }
+  stages {
+    stage ('Checking out branch and setting up environment') {
+      steps {
+        script {
+          checkout([$class: 'GitSCM', branches: [[name: '${CR_BRANCH}']], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'CleanCheckout']], submoduleCfg: [], userRemoteConfigs: [[url: 'git@github.com:skillz/3d-cave-runner.git']]])
+          configFileProvider([configFile(fileId: 'd355cd90-e5a7-47d3-84f0-d14d5acda8c3', targetLocation: '../../../../../../opsadmin/.gradle/gradle.properties'), configFile(fileId: 'd355cd90-e5a7-47d3-84f0-d14d5acda8c3', targetLocation: 'Android/3D Cave Runner - Android Studio/local.properties')]) {}
```

Comment:
Had to set targetLocation for grade.property to `'../../../../../../opsadmin/.gradle/gradle.properties'` because the configFileProvider command seems to start at the project root directory and theres not way to change it.

Reply:
is `../../../../../../opsadmin/.gradle/gradle.properties` the same as `~/.gradle/gradle.properties`?

Reply:
No it ends up being `/Users/Shared/jenkins-slave/workspace/SDK/Production%20Apps/CaveRunner/~/.gradle/gradle.properties` if I set the target location to `~/.gradle/gradle.properties` because the `configFileProvider` command seems to append the directory of the Jenkins build before the targetLocation instead of reading `~/` as home.